vowpalwabbit/cbify.cc

```diff
@@ -145,7 +147,13 @@ void cbify_adf_data::copy_example_to_adf(parameters& weights, example& ec)
     for (features& fs : eca)
     {
       for (feature_index& idx : fs.indicies)
-      { idx = ((((idx >> ss) * 28904713) + 4832917 * (uint64_t)a) << ss) & mask; }
+      {
+        auto rawidx = idx >> ss;
+        rawidx -= rawidx % (num_actions * increment);
+        rawidx += a * increment;
+        rawidx <<= ss;
+        idx = rawidx & mask;
+      }
```

Comment:
Can you explain this change?

Reply:
to avoid collisions between same name features in an example translated to multi_ex, each action is getting unique features even if they come from the same feature name (the input)